### PR TITLE
Correctly handle Biber versioning

### LINKED ISSUE #146: Cannot download a specific version of biber
Hi Manuel,
first of all, I'd like to preface this issue by thanking you for your OSS contribution ❤️. Your Action has been instrumental in significantly improving my CI build times.

Unfortunately, I ran into some problems with regards to Biber. The following log output was generated inside [act](https://github.com/nektos/act) btw:

```
[release-pls/build] ⭐  Run Setup Tectonic
INFO[0001]   ☁  git clone 'https://github.com/wtfjoke/setup-tectonic' # ref=v1
[release-pls/build]   🐳  docker cp src=/Users/pr1metine/.cache/act/wtfjoke-setup-tectonic@v1/ dst=/var/run/act/actions/wtfjoke-setup-tectonic@v1/
[release-pls/build]   🐳  docker exec cmd=[mkdir -p /var/run/act/actions/wtfjoke-setup-tectonic@v1/] user= workdir=
[release-pls/build]   🐳  docker exec cmd=[node /var/run/act/actions/wtfjoke-setup-tectonic@v1/dist/index.js] user= workdir=
[release-pls/build]   💬  ::debug::Finding releases for Tectonic version latest
[release-pls/build]   💬  ::debug::Getting build for Tectonic version 0.8.0: linux
[release-pls/build]   💬  ::debug::Release: {"id":51168607,"name":"tectonic 0.8.0","version":"0.8.0","tagName":"tectonic@0.8.0","assets":[{"name":"tectonic-0.8.0-arm-unknown-linux-musleabihf.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-arm-unknown-linux-musleabihf.tar.gz"},{"name":"tectonic-0.8.0-i686-unknown-linux-gnu.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-i686-unknown-linux-gnu.tar.gz"},{"name":"tectonic-0.8.0-mips-unknown-linux-gnu.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-mips-unknown-linux-gnu.tar.gz"},{"name":"tectonic-0.8.0-x86_64-apple-darwin.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64-apple-darwin.tar.gz"},{"name":"tectonic-0.8.0-x86_64-pc-windows-gnu.zip","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64-pc-windows-gnu.zip"},{"name":"tectonic-0.8.0-x86_64-pc-windows-msvc.zip","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64-pc-windows-msvc.zip"},{"name":"tectonic-0.8.0-x86_64-unknown-linux-gnu.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64-unknown-linux-gnu.tar.gz"},{"name":"tectonic-0.8.0-x86_64-unknown-linux-musl.tar.gz","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64-unknown-linux-musl.tar.gz"},{"name":"tectonic-0.8.0-x86_64.AppImage","url":"https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64.AppImage"}]}
[release-pls/build]   💬  ::debug::Downloading Tectonic from https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64.AppImage
[release-pls/build]   💬  ::debug::Downloading https://github.com/tectonic-typesetting/tectonic/releases/download/tectonic%25400.8.0/tectonic-0.8.0-x86_64.AppImage
[release-pls/build]   💬  ::debug::Destination /tmp/c6807eda-b589-4909-bdbd-6553215dfe2d
[release-pls/build]   💬  ::debug::download complete
[release-pls/build]   💬  ::debug::Extracting Tectonic
[release-pls/build]   💬  ::debug::Moved Tectonic from /tmp/c6807eda-b589-4909-bdbd-6553215dfe2d to /tmp/1eab2310-8440-451f-841c-70c94e842072/tectonic
[release-pls/build]   💬  ::debug::Tectonic path is /tmp/1eab2310-8440-451f-841c-70c94e842072
[release-pls/build]   💬  ::debug::Biber version: 2.16
[release-pls/build]   💬  ::debug::Downloading Biber from https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Linux/biber-linux_x86_64.tar.gz/download
[release-pls/build]   💬  ::debug::Downloading https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Linux/biber-linux_x86_64.tar.gz/download
[release-pls/build]   💬  ::debug::Destination /tmp/d192df98-3c57-49fb-92b7-0cb681ee6230
[release-pls/build]   💬  ::debug::download complete
[release-pls/build]   💬  ::debug::Extracting Biber
[release-pls/build]   💬  ::debug::Checking tar --version
[release-pls/build]   💬  ::debug::tar (GNU tar) 1.30%0ACopyright (C) 2017 Free Software Foundation, Inc.%0ALicense GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.%0AThis is free software: you are free to change and redistribute it.%0AThere is NO WARRANTY, to the extent permitted by law.%0A%0AWritten by John Gilmore and Jay Fenlason.
| [command]/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /tmp/369bbafa-5b3c-4726-bb2e-c2bf905cbfcd -f /tmp/d192df98-3c57-49fb-92b7-0cb681ee6230
[release-pls/build]   💬  ::debug::Biber path is /tmp/369bbafa-5b3c-4726-bb2e-c2bf905cbfcd
[release-pls/build]   ✅  Success - Setup Tectonic
```

When specifying a `biber-version`, your GitHub Action always downloads the `current` version regardless of the specified `biber-version`:

```
[release-pls/build]   💬  ::debug::Biber version: 2.16
[release-pls/build]   💬  ::debug::Downloading Biber from https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Linux/biber-linux_x86_64.tar.gz/download
[release-pls/build]   💬  ::debug::Downloading https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Linux/biber-linux_x86_64.tar.gz/download
```

That's because Biber does not strictly follow the Semantic Versioning conventions. Thus, the specified `biber-version` i. e. `2.16` is labelled invalid and defaults to `current`:

https://github.com/WtfJoke/setup-tectonic/blob/eb5e5c2ff6887545ae527012b424672b28fc34e0/src/biber.ts#L7-L8

Appending a patch version to `biber-version` i. e. `2.16.0` results in an error:

```
[release-pls/build]   💬  ::debug::Biber version: 2.16.0
[release-pls/build]   💬  ::debug::Downloading Biber from https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/2.16.0/binaries/Linux/biber-linux_x86_64.tar.gz/download
[release-pls/build]   💬  ::debug::Downloading https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/2.16.0/binaries/Linux/biber-linux_x86_64.tar.gz/download
[release-pls/build]   💬  ::debug::Destination /tmp/dd3d23c4-b1c0-40fc-a1dd-fc1fff58759b
[release-pls/build]   💬  ::debug::Failed to download from "https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/2.16.0/binaries/Linux/biber-linux_x86_64.tar.gz/download". Code(404) Message(Not Found)
[release-pls/build]   ❗  ::error::Error: Unexpected HTTP response: 404
[release-pls/build]   ❗  ::error::Unexpected HTTP response: 404
[release-pls/build]   ❌  Failure - Setup Tectonic
Error: exit with `FAILURE`: 1
```

I should be able to fix this issue on my own. I am going to open a PR for this soon.

### PR DESCRIPTION
**Pull request by pr1metine**
Fixes #146. 

`semver` continues to parse user input. With these commits, the user provided `biber-version` is first coerced into a SemVer object. Then, the major and minor version numbers are returned as a string e. g. `MAJOR.MINOR`.

If `semver` cannot turn the user-input `biber-version` into a valid SemVer object, this Action will fallback to the latest `current` version.

If a patch version is specified and non-zero, the patch version will be returned alongside the major and minor version e. g. `MAJOR.MINOR.PATCH`.

### CODE REVIEW

File: src/biber.ts [hunk 1]
```diff
@@ -2,10 +2,26 @@ import * as core from '@actions/core'
...
+      `Invalid biber version: ${biberVersion}. Defaulting to latest version`
+    )
+    return 'current'
+  }
+  if (biberSemVer.patch !== 0) {
+    return biberSemVer.version
+  }
+
+  return `${biberSemVer.major}.${biberSemVer.minor}`
+}
```
Comment (WtfJoke) [inline src/biber.ts:21]: Could you please add some tests to this newly introduced method?

File: __tests__/biber.test.ts [hunk 2]
```diff
@@ -14,13 +15,20 @@ test('build download link', async () => {
...
+
 test('download specific biber version', async () => {
   const biberPath = await downloadBiber('2.15')
   const fileExtension = os.platform() == 'win32' ? '.exe' : ''
   const expectedBinaryPath = path.resolve(biberPath, 'biber' + fileExtension)
 
   expect(fs.existsSync(biberPath)).toBe(true)
   expect(fs.existsSync(expectedBinaryPath)).toBe(true)
+  expect(execFileSync(expectedBinaryPath, ['--version']).toString()).toMatch(
+    /2\.15/
```
Comment (WtfJoke) [inline __tests__/biber.test.ts:30]: Great addition <3 Could have avoided your bug report like that

**Review by WtfJoke — COMMENTED** [review]
Thanks a lot for your PR.
Looks good so far. 👍  After adding some tests to the newly added method, im happly merging that!

File: src/biber.ts (hunk 1 above)
Comment (pr1metine) [reply src/biber.ts]: Ok, ty for your feedback. I addressed your suggestions in the latest commit

**Review by WtfJoke — APPROVED** [review]
Thanks you for your contribution :)
LGTM 🦫

**Comment by pr1metine** [comment]
Nice 😄. Could you also make a release so that I can profit from that fix as soon as possible?

**Pull request merged**